flang/lib/Lower/ConvertExprToHLFIR.cpp

```diff
@@ -1696,18 +1696,17 @@ class HlfirBuilder {
     // required chains of hlfir.designate to address the parent components so
     // that the StructureConstructor can later be lowered by addressing these
     // parent components if needed. Note: the front-end orders the components in
-    // structure constructors. The code below relies on the component to appear
-    // in order.
```

Comment:
Why was that assumption broken by the fact that the parent type was from a module in a different source file? It is true if the module is defined in the same source file.